[Gavino3o] iP

Gavino3o opened the pull request:
# Bob
> _We don't make mistakes, just happy little accidents._ - Bob Ross

**Bob is a Chatbot** that can help free your mind of having to remember things you need to do!
It is:
- Text-Based
- Easy to Learn
- _Fast_

If you are interested in Bob, here are some steps to get Bob up and running in your PC!
1. Download it from [here](https://github.com/Gavino3o/ip/releases).
1. Double-click the JAR file.
1. Start telling Bob what tasks to add.
1. Let Bob help you manage your tasks. 😁 

Bob is completely FREE!

## Features:
- [x] Add and delete todos, deadlines and events
- [x] List down your tasks
- [x] Search for tasks with a keyword
- [x] Mark and unmark your tasks as completed
- [ ] Search for tasks on a specific date (coming soon)
- [ ] Reminders (planned for future)

## Basic Syntax:
- todo \<description\>
- deadline \<description\> /by \<due_date\>
- event \<description\> /from \<start_date\> /to \<end_date\>
- list
- mark \<task_number\>
- unmark \<task_number\>
- bye

## 

If you are a Java Programmer, you can use it to practice Java too. Here is the `main` method:

```Java
    public static void main(String[] args) {
        new Bob("data/", "Bob.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Gavino3o committed:
Rename bot and modify greet and exit messages

Gavino3o committed:
Add echo functionality and refactor print messages

Gavino3o committed:
Add task storing and list task

Gavino3o committed:
Add Mark as Done functionality

Gavino3o committed:
Add Todos, Deadlines and Events

Gavino3o committed:
Implement initial testing

Gavino3o committed:
Add initial error handling for incorrect user inputs

Gavino3o committed:
Refactor response process to use ArrayList and recursion

Gavino3o committed:
Add delete task functionality

Gavino3o committed:
Refactor response processing to use enums

Gavino3o committed:
Update automated test to include delete

Gavino3o committed:
Fix incorrect command exception handling

Gavino3o committed:
Rename files from Duke to Bob

Gavino3o committed:
Create packages and reallocate relevant files

Gavino3o committed:
Add task saving functionality

Gavino3o committed:
Update .gitignore to ignore local data storage file

Gavino3o committed:
Merge branch 'branch-level-7'

Gavino3o committed:
Add Date recognition for Deadlines and Events

Gavino3o committed:
Update input test to expect LocalDate format pattern

Gavino3o committed:
Update .gitignore to ignore data directory in text-ui-test

Gavino3o committed:
Merge branch 'branch-Level-8'

Gavino3o committed:
Implement initial TaskList and related Exceptions

Gavino3o committed:
Implement initial Command Classes

Gavino3o committed:
Implement Parser, StorageFile and related Exceptions

Gavino3o committed:
Refactor Parser parse method for readability

Gavino3o committed:
Implement TextUi to print Command and Error messages

Gavino3o committed:
Refactor Bob.java to use newly decoupled classes

Gavino3o committed:
Bugfix for incorrect BobInvalidTaskNumberException message in TaskList

Gavino3o committed:
Merge branch 'A-MoreOOP'

Gavino3o committed:
Create and reallocate all packages and classes in new Bob package

Gavino3o committed:
Merge branch 'A-Packages'

Gavino3o committed:
Merge branch 'add-gradle-support' of https://github.com/Gavino3o/ip

Gavino3o committed:
Add JUnit test for Parser and TaskList

Gavino3o committed:
Merge branch 'A-JUnit'

Gavino3o committed:
Add JavaDoc comments to the code

Gavino3o committed:
Fix line spacing in BobArgumentException

Gavino3o committed:
Merge branch 'branch-A-JavaDoc'

Gavino3o committed:
Fix switch case indentation according to coding standard

Gavino3o committed:
Merge branch 'branch-A-CodingStandard'

Gavino3o committed:
Implement Find Command

Gavino3o committed:
Fix missing param descriptions in JavaDoc

Gavino3o committed:
Merge branch 'branch-Level-9'

Gavino3o committed:
Fix ordering and spacing of import statements

Gavino3o committed:
Add JUnit testing for delete and find command

Gavino3o committed:
Add CheckStyle and fix CheckStyle violations

Gavino3o committed:
Add blank line skipping when loading Tasks

Gavino3o committed:
Fix text-ui-test incorrect file path

Gavino3o committed:
Merge branch 'branch-A-CheckStyle'

Gavino3o committed:
Add TextGenerator and Refactor Bob.java for new GUI

Gavino3o committed:
Update build.gradle to include JavaFX v17.0.7

Gavino3o committed:
Implement initial skeleton of GUI

Gavino3o committed:
Implement basic styling for GUI skeleton

Gavino3o committed:
Add JavaDocs for controllers and Main

Gavino3o committed:
Merge branch 'branch-Level-10'

qyaner commented on src/main/java/bob/command/AddCommand.java:
```diff
@@ -0,0 +1,76 @@
+import bob.task.TaskList;
+import bob.task.Todo;
+import bob.ui.TextGenerator;
+import bob.ui.TextUi;
```
Good job on importing all the packages individually!

qyaner commented on src/main/java/bob/task/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+     * @param description Text description of Deadline Task
+     * @param by Due date of Deadline task
+     */
+    public Deadline(String description, String by) {
```
maybe can replace "by" with "dueDate" to make it more obvious?

qyaner commented on src/main/java/bob/parser/Parser.java:
```diff
@@ -0,0 +1,173 @@
+            argument = inputSplit[1];
+        }
+
+        switch (commandType) {
+        case BYE:
+            return new ExitCommand();
+        case LIST:
+            return new ListCommand();
+        case MARK:
+        case UNMARK:
+            return Parser.parseMarkCommand(commandType, argument);
+        case TODO:
+            return Parser.parseTodoCommand(commandType, argument);
+        case DEADLINE:
+            return Parser.parseDeadlineCommand(commandType, argument);
+        case EVENT:
+            return Parser.parseEventCommand(commandType, argument);
+        case DELETE:
+            return Parser.parseDeleteCommand(argument);
+        case FIND:
+            return Parser.parseFindCommand(argument);
+        case INVALID:
+            throw new BobInvalidCommandException("I'm sorry! I don't understand the command :(");
+        default:
+            throw new BobInvalidCommandException("I'm sorry! Something went wrong when processing your command :(");
+        }
+    }
```
naming sounds good in general, very clear and concise

qyaner commented on src/main/java/bob/command/Command.java:
```diff
@@ -0,0 +1,40 @@
+ * It can be stored an executed at a later instance or be reused.
+ */
+public abstract class Command {
+    /**
+     * Executes the current command and can modify the current state of the task list or
+     * print out messages.
+     *
+     * @param taskList Current list of tasks
+     * @param storageFile Storage file reader and writer
+     * @param ui User Interface reader and writer
+     * @throws BobException if the command is constructed with invalid arguments
+     */
```
Overall, I think the Javadocs are done well, and the descriptions provided are easy to understand and provide the important information for the class/methods.

qyaner reviewed (commented):
LGTM! Good job on following the coding standards!  👍

Gavino3o committed:
Rename 'by' to 'dueDate' in Deadline

WinSheng1 commented on src/main/java/bob/command/FindCommand.java:
```diff
@@ -0,0 +1,37 @@
+ * The FindCommand Class encapsulates logic that can be executed
+ * to find tasks with descriptions corresponding to a given keyword.
+ */
+public class FindCommand extends Command {
+
+    private String keyword;
```
Some of your classes have a spacing in between the class and fields whereas other doesn't. Keep it standardised.

WinSheng1 reviewed (commented):
Overall, the code was very well done and I will not be nitpicking for mistakes. Apart from line seperations which should be kept the same throughout, your code is generally easy to read, with variables and methods being named well. Javadocs is also done well with substantial description to explain the classes and methods. The code is well-structured and avoids deep nesting, making it easier to follow and understand. Keep up the good work!

Gavino3o committed:
Add basic null value assertions

There are several parts of the program flow that will be
affected if a null value is assigned instead of its intended
type of value.

Adding assertions to check for potential null values can help
minimise or prevent bugs.

Assert statements has been added with specific messages for
ease of identifying.

Gavino3o committed:
Remove unused imports and exception throwing

Gavino3o committed:
Bugfix for incorrect save file line reading

Gavino3o committed:
Enable assertions for gradle build

Gavino3o committed:
Merge pull request #1 from Gavino3o/branch-A-Assertions

Add assertions and minor bugfixes

Gavino3o committed:
Add gradle.yml

GitHub action will run the workflow (as per the .yml file)
everytime certain project events are triggered.

Examples include, when a PR is updated, or when the
master branch is updated.

Gavino3o committed:
Merge branch 'branch-A-CI'

Gavino3o committed:
Implement mark multiple and unmark multiple command

Commands to executes mass operations are widely seen in many text
based operations. The ability to execute mass operations can
increase the efficiency and improve the experience of the user.

The "markm" command and "unmarkm" command allows users to input
multiple evenly spaced integer numbers to mark the corresponding
tasks as done.

Gavino3o committed:
Implement delete multiple command

Gavino3o committed:
Merge pull request #2 from Gavino3o/branch-C-MassOps

Implement Mass Operation for mark, unmark and delete

Gavino3o committed:
Add guard clause for delete multiple command

Gavino3o committed:
Combine Multiple type commands with Single type

There exists code duplication and similar functionality between
markm, unmarkm, and deletem commands with their single type
counterparts. They should be removed to increase code quality
and abide by DRY principle.

Changes has been made so that mark, unmark, and delete commands
can handle mass operations.

Gavino3o committed:
Merge pull request #3 from Gavino3o/branch-A-CodeQuality

Refactor points of code for Code Quality

Gavino3o committed:
Modify Dialog Box Image to be a Circle

Gavino3o committed:
Add MainWindow minHeight and minWidth

Gavino3o committed:
Modify MainWindow to use GridPane

Gavino3o committed:
Modify background colour and text bubble

Gavino3o committed:
Merge branch 'branch-A-BetterGui'

Gavino3o committed:
Update README.md

Included basic user guide in README for user reference.

Gavino3o committed:
Merge pull request #5 from Gavino3o/branch-README

Update README.md

Gavino3o committed:
Update README.md

Gavino3o committed:
Update README.md

Fix typo within headings.

Gavino3o committed:
Merge pull request #6 from Gavino3o/branch-README

Update README.md

Gavino3o committed:
Fix incorrect deletion order for delete command

Gavino3o committed:
Merge branch 'branch-MassDeleteBugFix'

Gavino3o committed:
Add Ui.png to docs

Gavino3o committed:
Modify build.gradle for cross platform dependencies

Gavino3o committed:
Merge branch 'branch-BuildGradleFix'
